test.cpp

```diff
@@ -0,0 +1,11 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "test/catch.hpp"
+#include "main.h"
+
+TEST_CASE("detect set of values  from  specified range within input array of integers") {
```

Comment:
Test case needs to be defined based on the requirements. So there is scope to add more tests which reflect the given requirement. It could be taken as an improvement ...